Space-Zoologist/Assets/Scripts/UI/Store/SellingManager.cs

```diff
@@ -0,0 +1,130 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class SellingManager : MonoBehaviour
+{
+    GridSystem gridSystem = default;
+    TileSystem tileSystem = default;
+    [SerializeField] PauseManager PauseManager = default;
+    [SerializeField] MenuManager MenuManager = default;
+    [SerializeField] Inspector Inspector = default;
+    [SerializeField] PlayerBalance PlayerBalance = default;
+    [SerializeField] LevelDataReference LevelDataReference = default;
+    public bool IsSelling { get; private set; }
+
+    public void OnToggleSell()
+    {
+        if (!IsSelling)
+        {
+            this.PauseManager.TryToPause();
+            if (Inspector.IsInInspectorMode)
+            {
+                Inspector.CloseInspector();
+            }
+            if (MenuManager.IsInStore)
+            {
+                MenuManager.CloseStore();
+            }
+        }
+        IsSelling = !IsSelling;
+    }
+
+    public void StopSelling() {
+        IsSelling = false;
+    }
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        IsSelling = false;
+        gridSystem = FindObjectOfType<GridSystem>();
+        tileSystem = FindObjectOfType<TileSystem>();
+        // stop selling when store opens
+        EventManager.Instance.SubscribeToEvent(EventType.StoreOpened, () =>
+        {
+            this.IsSelling = false;
+        });
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        // Much taken from Inspector
+        if (IsSelling && Input.GetMouseButtonDown(0))
+        {
+            // Update animal locations
+            gridSystem.UpdateAnimalCellGrid();
+
+            Vector3 worldPos = Camera.main.ScreenToWorldPoint(Input.mousePosition);
+            Vector3Int cellPos = tileSystem.WorldToCell(worldPos);
+            TerrainTile tile = tileSystem.GetTerrainTileAtLocation(cellPos);
+
+            //Debug.Log($"Mouse click at {cellPos}");
+
+            GridSystem.CellData cellData;
+
+            // Handles index out of bound exception
+            if (gridSystem.isCellinGrid(cellPos.x, cellPos.y))
+            {
+                cellData = gridSystem.CellGrid[cellPos.x, cellPos.y];
+            }
+            else
+            {
+                // Debug.Log($"Grid location selected was out of bounds @ {cellPos}");
+                return;
+            }
+
+            if (cellData.ContainsMachine)
```

Comment:
Break up the logic in these if statements into more functions. We want our code to be easy to read so it's easier to maintain and work with.

Reply:
I split it into SellMachineOnCell(), SellFoodOnCell(), SellWallOnCell().